lib/pwwka/receiver.rb

```diff
@@ -79,5 +79,11 @@ def test_teardown
       channel_connector.connection_close
     end
 
+    def prepare_payload(payload)
```

Comment:
A couple thoughts:

1 - we probably want the default behavior when JSON can't be parsed to continue to be an exception gets raised, and then do this only if there is a configuration option set not to (we rely on the existing behavior   for invalid JSON and likely others do, too)
2 - If this is for parsing non-JSON, this means that every non-JSON message has to get an exception raised.  That might be problematic.

Reply:
Thanks for the feedback.

I like the idea in the first point. I will go back to the drawing board to figure out how to get this done. I welcome tips and pointers with open arms.

For the second, since I catch all exceptions which come from parsing, should this be an issue? Or are we concerned about the performance cost which comes form raising exceptions?

Reply:
More thinking about the perf cost of having an exception raised on every message. I don't have a good sense of the magnitude of that, though

Reply:
Maybe call this `parse_payload`?